src/crypto/equihash.cpp

```diff
@@ -34,11 +34,105 @@ int Equihash<N,K>::InitialiseState(eh_HashState& base_state)
     memcpy(personalization+12, &le_K, 4);
     return crypto_generichash_blake2b_init_salt_personal(&base_state,
                                                          NULL, 0, // No key.
-                                                         N/8,
+                                                         (512/N)*N/8,
                                                          NULL,    // No salt.
                                                          personalization);
 }
 
+void GenerateHash(const eh_HashState& base_state, eh_index g,
+                  unsigned char* hash, size_t hLen)
+{
+    eh_HashState state;
+    state = base_state;
+    unsigned char array[sizeof(eh_index)];
+    eh_index lei = htole32(g);
+    memcpy(array, &lei, sizeof(eh_index));
+    crypto_generichash_blake2b_update(&state, array, sizeof(eh_index));
```

Comment:
I vaguely recall there being some reason why I did it this way (the code here was simply shifted from the `StepRow` constructor), but it completely escapes me at present. I'll poke around in the commit history and see if I can remember...